src/d/actor/d_a_shand.cpp

```diff
@@ -4,91 +4,488 @@
 }
 
 /* 0000225C-00002264       .text daShand_IsDelete__FP11shand_class */
 static BOOL daShand_IsDelete(shand_class*) {
-    /* Nonmatching */
+    return TRUE;
 }
 
 /* 00002264-000022D4       .text daShand_Delete__FP11shand_class */
-static BOOL daShand_Delete(shand_class*) {
-    /* Nonmatching */
+static BOOL daShand_Delete(shand_class* actor) {
+    dComIfG_resDelete(&actor->mPhs, "Shand");
+    if(actor->mHasHIO){
+        hio_set = false;
+        mDoHIO_deleteChild(l_HIO.mId);
+    }
+
+    return TRUE;
 }
 
 /* 000022D4-00002360       .text useHeapInit__FP11shand_class */
-void useHeapInit(shand_class*) {
-    /* Nonmatching */
+static BOOL useHeapInit(shand_class* actor) {
+    const int res_index = ((fopAcM_GetParam(actor) & 0xff) == 53) ? 4 : 3;
+    ResTIMG* res = static_cast<ResTIMG*>(dComIfG_getObjectRes("Shand", res_index));
+    
+    if(actor->mLineMat.init(1, 20, res, 1) == FALSE){
+        return FALSE;
+    }
+    else{
+        return TRUE;
+    }
 }
 
 /* 00002360-00002380       .text daShand_solidHeapCB__FP10fopAc_ac_c */
-static BOOL daShand_solidHeapCB(fopAc_ac_c*) {
-    /* Nonmatching */
+static BOOL daShand_solidHeapCB(fopAc_ac_c* actor) {
+    return useHeapInit(static_cast<shand_class*>(actor));
 }
 
 /* 00002380-00002630       .text daShand_Create__FP10fopAc_ac_c */
-static cPhs_State daShand_Create(fopAc_ac_c*) {
-    /* Nonmatching */
+static cPhs_State daShand_Create(fopAc_ac_c* f_this) {
+    static dCcD_SrcCyl tg_cyl_src = {
+        0,
+        0, 
+        0,
+        0,
+        AT_TYPE_ALL - (AT_TYPE_WATER | AT_TYPE_UNK20000 | AT_TYPE_WIND | AT_TYPE_UNK400000 | AT_TYPE_LIGHT), // 0xff1dfeff 
+        cCcD_TgSPrm_Set_e | cCcD_TgSPrm_IsEnemy_e,
+        cCcD_CoSPrm_Set_e | cCcD_CoSPrm_IsPlayer_e | cCcD_CoSPrm_VsGrpAll_e, 
+        0, 
+        0, 
+        0,
+        0,
+        0,
+        0,
+        0,
+        0,
+        0, 
+        0,
+        0,
+        0,
+        0,
+        0,
+        15.0f, 
+        200.0f
+    };
+
+    static dCcD_SrcSph bm_sph_src = {
+        0,
+        0,
+        0,
+        5,
+        AT_TYPE_ALL - (AT_TYPE_WATER | AT_TYPE_UNK20000 | AT_TYPE_WIND | AT_TYPE_UNK400000 | AT_TYPE_LIGHT), 
+        cCcD_TgSPrm_Set_e | cCcD_TgSPrm_IsEnemy_e,
+    0,
+        0, 
+        0, 
+        0,
+        0,
+        0,
+        0,
+        0,
+        0,
+        0, 
+        0,
+        0,
+        0,
+        0,
+        0,
+        80.0f
+    };
+
+    fopAcM_SetupActor(f_this, shand_class);
+    shand_class* s_this = static_cast<shand_class*>(f_this);
+    int ret = dComIfG_resLoad(&s_this->mPhs, "Shand");
+    if(ret == cPhs_COMPLEATE_e){
+        if(fopAcM_entrySolidHeap(s_this, daShand_solidHeapCB, 0x1040) != false){
+            s_this->health = 2;
+            s_this->mExecuteCount = cM_rndF(10000.0f);
+            s_this->field_318 = 10;
+            s_this->field_2BA = 1;
+            s_this->field_2F0 = 1.0f;
+            if((fopAcM_GetParam(s_this) & 0xff) == true)
```

Comment:
```suggestion
            if((fopAcM_GetParam(s_this) & 0xff) == 1)
```
This param probably isn't a bool since it's being compared to 53 elsewhere in the code